Action test

Dax233 opened the pull request:
## Sourcery 总结

重构核心启动，使用声明式 ServiceBuilder 进行依赖注入，并替换旧的 CoreSystemInitializer 为简化的异步主入口点。重新设计 QQBuilder，通过通用模板和动态白名单统一动作事件构建。彻底改造 CoreLogic 的思考循环，使其在主要思考生成和中断哨兵之间竞争，并持久化中断上下文。将提示构建模块化为可重用的 PromptComponents，带有动态系统和用户块，从而实现模式驱动的控制和操作。增强 ChatSessionManager，采用基于栈的焦点模型（push/pop/swap/teleport/jump）、动态会话激活/停用以及健壮的摘要持久化。改进 DefaultMessageProcessor，以检测和路由 send_message “echo” 事件并分离事件持久化。精简 ActionHandler，以支持核心 web_search，统一动作执行流程，并消除决策处理中的副作用。改造 PendingActionManager, CoreWebsocketServer, UnreadInfoService 和 PersonStorageService，以支持自我资料注册、带指数退避的适配器检查以及顶层平台和会话摘要。整合日志设置，用于每日/每月维护、居中别名格式化和异步文件轮换。更新运行脚本和 README，以反映新的架构和用法。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery
重构核心启动，使用声明式 ServiceBuilder 进行依赖注入，并采用异步主入口；简化并模块化平台构建器和提示词构建；增强焦点管理和消息/动作处理；全面改进日志管理；并更新文档。

New Features:
- 引入 ServiceBuilder 和依赖注入，用于核心系统初始化
- 添加基于栈的焦点控制命令 (push_focus, pop_focus, swap_focus, teleport_focus, jump_to_history)
- 通过动态白名单和模板在 QQBuilder 中启用通用动作处理

Enhancements:
- 重构主逻辑以使用异步 main 和 wire_dependencies 实现更简洁的启动
- 将 PromptBuilder 模块化为可重用的 PromptComponents，带有动态系统/用户模板
- 增强 DefaultMessageProcessor 以检测并将自回显消息路由到 ChatSession
- 简化 ActionHandler 并添加直接的 web_search 动作支持
- 改进 PendingActionManager，支持回显暂存、主动会话文档创建和更丰富的响应处理
- 升级 CoreWebsocketServer，支持连接时检查重试、工具平台的轻量级身份注册和平台摘要集成
- 改进日志配置，以执行启动清理、每日压缩、每月归档和居中别名格式化
- 增强数据库服务，使用高性能 AQL 实现最近活跃会话、自账户查询和思维动作结果持久化

Documentation:
- 全面修订 README 文档，以记录新架构、配置工作流和使用步骤

<details>
<summary>Original summary in English</summary>

## Sourcery 总结

重构启动过程，使用异步依赖注入容器，简化核心入口，并引入 ServiceBuilder。全面改进 QQBuilder 和平台构建器，使用通用事件模板。将提示生成模块化，并添加基于栈的焦点控制，以实现动态会话管理。统一通用动作和 web_search 动作的处理，改进中断驱动的 CoreLogic 循环，并简化消息处理。增强 PendingActionManager 和未读摘要查询。添加指数退避适配器检查和全面日志归档。更新文档以反映新架构和工作流程。

新功能：
- 实现基于异步 DI 的启动，使用 ServiceBuilder，并用简化的异步入口点替换 CoreSystemInitializer
- 重新设计 QQBuilder，支持通用和特殊动作处理器，并引入动态白名单以将动作转换为标准事件
- 将提示构建模块化为 PromptComponents，支持动态系统/用户块和模式驱动的控制/动作
- 引入基于栈的意识控制命令 (push_focus, pop_focus, swap_focus, teleport_focus, jump_to_history) 和动态会话激活/停用
- 在 ActionHandler 中添加核心层级的 web_search 动作支持，使用专用的 LLM 代理
- 实现指数退避适配器检查，并在适配器连接时进行轻量级身份注册
- 集成每日日志自动压缩和每月归档，支持居中别名格式和异步文件轮转

改进：
- 将 CoreLogic 重构为主思想生成与智能中断哨兵之间的竞争模式
- 改进 DefaultMessageProcessor，使其能检测并路由自回显的 send_message 事件，并分离事件持久化
- 简化 ActionHandler，以统一平台和核心动作，消除副作用，并使用通用事件构建器
- 改进 PendingActionManager 和会话查询，支持详细的未读摘要 AQL、激励上下文和健壮的数据库更新
- 重构 ChatSessionManager，实现并发安全的焦点历史、持久化的中断上下文和改进的会话生命周期
- 简化 UnreadInfoService，使用核心会话服务获取摘要并更新自机器人 ID 映射
- 总体清理平台构建器基础接口和服务方法，移除废弃代码并提高一致性

文档：
- 更新 README，文档化新的 DI 启动、三层注意力模型、配置模板和使用说明

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Refactor startup to use asynchronous DI container, simplify core entry, and introduce ServiceBuilder. Overhaul QQBuilder and platform builders with generic event templates. Modularize prompt generation and add stack-based focus controls for dynamic session management. Unify action handling with generic and web_search actions, improve interruption-driven CoreLogic loop, and streamline message processing. Enhance PendingActionManager and unread-summary queries. Add exponential backoff adapter inspection and comprehensive log archival. Update documentation to reflect the new architecture and workflows.

New Features:
- Implement asynchronous DI-based startup using ServiceBuilder and replace CoreSystemInitializer with a streamlined async entry point
- Redesign QQBuilder with generic and special action handlers and dynamic whitelist for translating actions into standard Events
- Modularize prompt construction into PromptComponents with dynamic system/user blocks and schema-driven controls/actions
- Introduce stack-based consciousness control commands (push_focus, pop_focus, swap_focus, teleport_focus, jump_to_history) and dynamic session activation/deactivation
- Add core-level web_search action support in ActionHandler using a dedicated LLM agent
- Implement exponential backoff adapter inspection and lightweight identity registration on adapter connect
- Integrate automatic daily log compression and monthly archiving with centered alias formatting and async file rotation

Enhancements:
- Refactor CoreLogic into a race between main thought generation and an intelligent interruption sentinel
- Improve DefaultMessageProcessor to detect and route self-echo send_message events and separate event persistence
- Streamline ActionHandler to unify platform and core actions, eliminate side effects, and use generic event builders
- Revamp PendingActionManager and conversation queries to support detailed unread-summary AQL, motivational context, and robust DB updates
- Restructure ChatSessionManager for concurrency-safe focus history, persistent interruption context, and improved session lifecycle
- Simplify UnreadInfoService to use core conversation service for summaries and update self-bot IDs mapping
- General cleanup of platform builders base interface and service methods to remove dead code and improve consistency

Documentation:
- Update README to document new DI startup, three-layer attention model, configuration templates, and usage instructions

</details>

</details>

</details>

smartmita committed:
确保异常能被捕获

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

smartmita committed:
qq builder规范

Dax233 committed:
删除重复函数, 统一使用SegBuilder

smartmita committed:
推

smartmita committed:
重构进度：
核心（Core）进度约 85%，
适配器（Adapter）进度约 30%。

smartmita committed:
终于可以跑了

smartmita committed:
整理/修复

smartmita committed:
修复

smartmita committed:
规范化注释

smartmita committed:
无法测试，只能推送

Dax233 committed:
修复无法启动问题

Dax233 committed:
ruff

smartmita committed:
快收尾完了，还不能拉

smartmita committed:
修复获取可选字段描述重复

smartmita committed:
杀杀杀（优化/修复）

smartmita committed:
修复prompt格式错误

smartmita committed:
再次修复prompt显示错误

smartmita committed:
优化，加入action分类逻辑

Dax233 committed:
Merge branch 'dev' into action_test

smartmita committed:
修复action的格式问题

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

smartmita committed:
修复未读块

smartmita committed:
写文档

smartmita committed:
测试性：将自身思想等放入system prompt

smartmita committed:
prompt跟上

Dax233 committed:
修复因未读服务不存在conn_manager导致无法运行的情况

Dax233 committed:
修复事件不持久化问题

Dax233 committed:
ruff

Dax233 committed:
添加缺失逻辑

smartmita committed:
修改发送消息的schema

smartmita committed:
style: Ruff 自动应用代码修复与格式化 ✨

smartmita committed:
补充缺失逻辑

Dax233 committed:
统一qq作为平台id

smartmita committed:
修改prompt试图解决格式不一致的问题

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

smartmita committed:
up

smartmita committed:
优化prompt

Dax233 committed:
修复schema输出问题, 删除了不支持但是不报错的可恶的oneOf

Dax233 committed:
修复发言类型错误

smartmita committed:
修复了以下bug：
1，action成功但依然显示参数错误
2，回声超时
3，安检失败
4，其它中途遇到的忘了的bug
试图修复以下bug，但是未完成：
1，机器人自身消息不显示在聊天记录里
2，发消息的动机没有正确附带

Dax233 committed:
修复回声匹配失败

Dax233 committed:
修复聊天记录缺失问题

Dax233 committed:
ruff

Dax233 committed:
新增消息数量计数重置

Dax233 committed:
fix. 采用健壮性更强的 loguru 自带的日志整理
TODO: 目前 Debug 层级为硬编码显示，之后再将其提取出环境配置中进行配置

smartmita committed:
修复聊天记录自身消息格式

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
完成 logger 修复

Dax233 committed:
修复核心循环出错

Dax233 committed:
修复未读消息记录错误的问题

smartmita committed:
修正动作想要的位置，试图简单方法改善llm对外部消息的误解

smartmita committed:
style: Ruff 自动应用代码修复与格式化 ✨

smartmita committed:
微调prompt

Dax233 committed:
尝试重构IIS上下游逻辑, zuo'deng

Dax233 committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

smartmita committed:
修复循环逻辑，修复retuen逻辑，修复获取自身信息逻辑，移除会导致报错的旧逻辑

Dax233 committed:
注释掉一个刷屏的日志

smartmita committed:
推一个调试logger

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
注释掉一个刷屏日志

smartmita committed:
大概率没有完全修好

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
修复iis

Dax233 committed:
修复发言中上下文变化

Dax233 committed:
Feat. Add termux_builder.py

Dax233 committed:
Better. Adjust prompt

smartmita committed:
1，修复unread块
2，修复get list后对没有互动过的会话无法进入

smartmita committed:
1,修复了Bug：我们通过在 ChatSessionManager 中追踪 previous_focus_path，确保了即使返回到核心层，也能生成有意义的描述，彻底告别 "这个地方"。
2,实现了增强功能：现在，每当AI的注意力发生切换，internal_info 块中都会生成一条非常清晰、连贯的 "从 Y 来到 X" 的描述，极大地增强了上下文的连续性。
3,优化了代码结构：我们将生成描述的复杂逻辑收敛到了 ChatSessionManager 中，InternalInfoBuilder 的职责变得更加单一和清晰，符合高内聚、低耦合的设计原则。

Dax233 committed:
上线安检功能解耦

Dax233 committed:
人物登记功能解耦

Dax233 committed:
尝试修复工具类ada初始化错误

Dax233 committed:
修复回调错误

Dax233 committed:
动作系统与ada解耦

Dax233 committed:
解耦action与平台, 新增面对离线平台的优雅处理

smartmita committed:
重构，目标文档：AIcarus 认知导航系统 v2.0

先推 有bug再说

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
新增轻量化身份登记函数, 用于工具类ada的使用

Dax233 committed:
补充缺少的传入参数

Dax233 committed:
删除无用函数

smartmita committed:
1：修复了数据类型不匹配的根本问题，通过增加一个步骤来从字典中提取字符串，确保了 .split() 操作的安全性。
2：修改了读取数据的一方，没有改ChatSessionManager 中存储数据的逻辑。使其可以继续用于导航日志等功能。
3：通过 isinstance 检查，代码现在更能抵抗因 focus_history 中可能存在的 None 或其他非字典类型值而导致的潜在崩溃。

Dax233 committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
Core_Builder中采用更准确的描述

smartmita committed:
增加健壮性，规避llm的不确定输出

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
ruff

Dax233 committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
style: Ruff 自动应用代码修复与格式化 ✨

smartmita committed:
现在很健壮了

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
优化jsonschema及其解析逻辑

Dax233 committed:
fixed

smartmita committed:
prompt优化

smartmita committed:
prompt优化

smartmita committed:
调换一下位置

smartmita committed:
1，通过修改 ChatSessionManager.deactivate_session，在停用会话时使用当前时间作为 last_processed_timestamp，确保了“已读”状态的准确性。
2，通过在 DefaultMessageProcessor.process_event 中添加逻辑，在处理非AI发送的新消息时，调用 session.reset_consecutive_bot_message_count()，确保了计数器的正确重置。

smartmita committed:
style: Ruff 自动应用代码修复与格式化 ✨

Dax233 committed:
typo

smartmita committed:
调整prompt

smartmita committed:
更加自然语言的描述

smartmita committed:
微调prompt

Dax233 committed:
builder通用化改造

Dax233 committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
删除部分过时文档

Dax233 committed:
修复会话层中无法解析核心层动作的问题

smartmita committed:
修复web search

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

smartmita committed:
再修

Dax233 committed:
ruff

smartmita committed:
小修web search字段小问题

smartmita committed:
联网搜索prompt优化

Dax233 commented on src/database/services/action_log_storage_service.py:
```diff
@@ -166,24 +107,47 @@ async def update_action_log_with_response(
 
-    async def get_recent_action_logs(self, limit: int = 10) -> list[dict[str, Any]]:
-        """获取最近的动作日志记录.
+    # =======================【 这 里 就 是 新 增 的 欲 望！】=======================
+    async def get_action_log_by_platform_message_id(self, message_id: str) -> dict[str, Any] | None:
```
你给出的建议实现非常棒，直接在服务初始化的时候检查并创建索引。不过，考虑到我们项目已经有了一套更集中的基础设施初始化流程，我有一个可能更好的点子.
我的计划是这样的：
集中定义 (Centralize Definition)：我们有一个管理所有数据库集合和索引定义的“圣地”——src/database/models.py 里的 CoreDBCollections 类。我会把这个新的索引定义也写在那里。这是所有数据库结构的唯一真实来源。
自动应用 (Automate Application)：我们的 ArangoDBConnectionManager 在程序启动时，会调用 ensure_core_infrastructure 方法，这个方法会读取 CoreDBCollections 里的所有定义，然后自动地去创建所有需要的集合和索引。
服务无感 (Service Agnostic)：ActionLogStorageService 本身不需要关心索引是否存在。它的职责是“使用”数据库，而不是“配置”数据库。基础设施的配置应该在程序启动的更高层面完成。

这种方式的好处是：
关注点分离：服务的代码只负责业务逻辑，数据库的结构定义和初始化逻辑是分开的。
配置集中化：所有数据库的“蓝图”都在一个文件里，以后要修改或添加索引一目了然。
一次性设置：索引的检查和创建只会在整个应用启动时执行一次，而不是每次实例化服务时都去检查，性能更好。

对此, 你怎么看?

smartmita replied:
@sourcery-ai 你能就你说的需要注意的两点以markdown格式写一份文档吗？

Dax233 committed:
全面增强代码健壮性与性能的综合小重构

Dax233 committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
注释掉刷屏log

smartmita committed:
修复AI对当前专注会话中的非消息事件（如通知、系统事件）无法做出即时反应的问题

Dax233 committed:
代码质量优化

Dax233 committed:
fix. "中断哨兵任务异常: cannot access local variable 'last_text_content_in_batch' where it is not associated with a value"

smartmita committed:
对临时会话的初步支持

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

smartmita committed:
补上unread的修改

Dax233 committed:
ruff

Dax233 commented:
@sourcery-ai dismiss

Dax233 commented:
@sourcery-ai review

Dax233 committed:
全面增强代码健壮性与性能的综合小重构

smartmita committed:
孤儿方法

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

smartmita committed:
style: Ruff 自动应用代码修复与格式化 ✨

smartmita committed:
让get list更加泛用

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

smartmita committed:
style: Ruff 自动应用代码修复与格式化 ✨

Dax233 committed:
修复iis, 暂时禁用summary

Dax233 committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
修复报错

Dax233 commented:
@sourcery-ai dismiss

Dax233 commented:
@sourcery-ai review

smartmita committed:
1，修复了逻辑缺陷：has_high_priority 的判断现在依赖于从 UnreadInfoService 传入的、经过“安检”确认的权威 bot_id 映射，确保了判断的准确性，无论是普通群聊、私聊还是临时会话。
2，遵循了单一事实来源原则：系统中关于“我是谁”的判断，其最终依据统一到了 PersonStorageService 和安检流程，避免了数据不一致导致的问题。

smartmita committed:
Merge branch 'action_test' of https://github.com/AIcarusDev/AIcarusCore into action_test

Dax233 committed:
修复iis
